operator: remove deprecated Azure cloud name flag

tklauser opened the pull request:
Commit b29389d50f97 ("Auto-detect Azure cloud name via IMDS") marked the
flag as deprecated in 1.10 and to be removed in 1.11.

qmonnet commented on Documentation/operations/upgrade.rst:
```diff
@@ -336,6 +336,8 @@ Removed Options
 * The deprecated ``hubble-ca-cert`` configmap has been removed. Use
   ``hubble-ca-secret`` secret instead.
+* The deprecated ``azure-cloud-name`` option for cilium-operator-azure was
+  deprecated in 1.10 and is now removed.
```
Nit: `The deprecated ... was deprecated in ...` sounds a bit redundant?

tklauser replied:
Indeed :smile: dropped the first one.

qmonnet replied:
thank you :)

qmonnet commented on pkg/ipam/allocator/azure/azure.go:
```diff
@@ -38,14 +37,10 @@ func (*AllocatorAzure) Start(ctx context.Context, getterUpdater ipam.CiliumNodeG
+	log.Debug("Retrieving Azure cloud name via Azure IMS")
+	azureCloudName, err := azureAPI.GetAzureCloudName(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("unsable to retrieve Azure cloud name: %w", err)
```
`unsable` -> Would you mind fixing the typo, please?

tklauser replied:
Fixed

qmonnet reviewed (approved):
Looks good thank you!
Two minor nits below, but they're non-blocking.

tklauser committed:
operator: remove deprecated Azure cloud name flag

Commit b29389d50f97 ("Auto-detect Azure cloud name via IMDS") marked the
flag as deprecated in 1.10 and to be removed in 1.11.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

ungureanuvladvictor reviewed (approved):
Thank you for this @tklauser !
